packages/gatsby-plugin-sharp/src/index.js

```diff
@@ -226,8 +226,7 @@ function queueImageResizing({ file, args = {} }) {
 
   const argsDigestShort = argsDigest.substr(argsDigest.length - 5)
 
-  const imgSrc = `/${file.internal
-    .contentDigest}-${argsDigestShort}.${fileExtension}`
+  const imgSrc = `/${file.name}-${argsDigestShort}.${fileExtension}`
```

Comment:
Oh, the filename needs the `contentDigest` of the original image still. That's how we ensure long-term caching because it's the unique "fingerprint" of the image.

Reply:
So should it be `${file.name}-${contentDigest}`?

Reply:
`${file.name}-${contentDigest}${argsDisgestShort}.${fileExtension}`